src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.HillClimbing.cs

```diff
@@ -13,9 +13,45 @@ internal partial class ClrThreadPool
         /// </summary>
         private partial class HillClimbing
         {
-            // Config values pulled from CoreCLR
-            // TODO: Move to runtime configuration variables.
-            public static HillClimbing ThreadPoolHillClimber { get; } = new HillClimbing(4, 20, 100 / 100.0, 8, 15 / 100.0, 300 / 100.0, 4, 20, 10, 200, 1 / 100.0, 200 / 100.0, 15 / 100.0);
+            private static readonly Lazy<HillClimbing> s_threadPoolHillClimber = new Lazy<HillClimbing>(CreateHillClimber, true);
+            public static HillClimbing ThreadPoolHillClimber => s_threadPoolHillClimber.Value;
+
+            private static int GetConfig(string configName, int defaultValue)
+            {
+                object config = AppContext.GetData(configName);
+                switch (config)
+                {
+                    case string str:
+                        if (int.TryParse(str, out int result))
```

Comment:
CoreCLR parses string-encoded integer values as hex. I suggest parsing strings prefixed with 0x as hex and those without the prefix as decimal. Also, since we know whether the value is signed or unsigned, we can parse the string with the appropriate type for the value, to allow or deny a sign.